File: mvptests/app/src/main/java/com/example/mvp_tests/MainActivity.java
package com.example.mvp_tests;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.List;

public class MainActivity extends AppCompatActivity implements MainScreenContract.View {
    MainScreenContract.Presenter presenter;

    boolean titleVisible;
    TextView title;
    TextView clientList;
    Button showTitleButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initialize();
        initializeListeners();
    }

    private void initialize() {
        titleVisible = true;
        title = findViewById(R.id.textViewTitle);
        clientList = findViewById(R.id.textViewClients);
        showTitleButton = findViewById(R.id.buttonShowTitle);

        presenter = new PresenterImpl(this, new ModelMockImpl());
        presenter.onViewCreated();
    }

    private void initializeListeners(){
        showTitleButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.onHideTitleButtonPressed();
            }
        });
    }

    @Override
    public void showClients(List<String> clients) {
        clientList.setText("");
        for (int i = 0; i < clients.size(); i++) {
            clientList.append(String.valueOf(clients.get(i)) + "\n");
        }
    }

    @Override
    public void showTitle() {
        title.setVisibility(View.VISIBLE);
        titleVisible = true;
    }

    @Override
    public void hideTitle() {
        title.setVisibility(View.INVISIBLE);
        titleVisible = false;
    }

    @Override
    public boolean isTitleVisible() {
        return titleVisible;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        presenter.onViewDestroy();
    }
}

File: music-player/app/src/main/java/com/example/musicplayer/MainActivity.kt
package com.example.musicplayer

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Button
import android.widget.TextView
import com.example.musicplayer.models.AudioManagerImpl
import com.example.musicplayer.models.PlayListImpl
import com.example.musicplayer.models.Song
import com.example.musicplayer.utils.Mp3Reader

class MainActivity : AppCompatActivity(), MediaPlayerContract.View {
    private var mSongTitle: TextView? = null
    private var mPlay: Button? = null
    private var mPause: Button? = null
    private var mPrevious: Button? = null
    private var mNext: Button? = null
    private var presenter: MediaPlayerContract.Presenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init()
        initViews()
        initListeners()
    }

    private fun init() {
        presenter = PresenterImpl(
            this,
            AudioManagerImpl(
                applicationContext, PlayListImpl(
                    arrayListOf(
                        mp3ToSong(R.raw.mpthreetest)
                    )
                )
            )
        )
    }

    private fun mp3ToSong(songId: Int): Song {
        val song = Song(songId)
        val mp3Data = Mp3Reader.loadFile(resources.openRawResource(songId))
        song.title = mp3Data.title
        song.author = mp3Data.author
        song.album = mp3Data.album
        song.year = mp3Data.year
        return song
    }

    private fun initViews() {
        mSongTitle = findViewById(R.id.textViewSongTitle)
        mPlay = findViewById(R.id.buttonPlay)
        mPause = findViewById(R.id.buttonPause)
        mPrevious = findViewById(R.id.buttonPrevious)
        mNext = findViewById(R.id.buttonNext)
    }

    private fun initListeners() {
        mPlay?.setOnClickListener { presenter?.onPlayButtonPressed() }
        mPause?.setOnClickListener { presenter?.onPauseButtonPressed() }
        mPrevious?.setOnClickListener { presenter?.onPreviousButtonPressed() }
        mNext?.setOnClickListener { presenter?.onNextButtonPressed() }
    }

    override fun showCurrentSongTitle(name: String) {
        mSongTitle?.text = name
    }

    override fun showPaused() {
        Log.d("STATUS_MEDIAPLAYER", "PAUSED")
    }

    override fun showPlaying() {
        Log.d("STATUS_MEDIAPLAYER", "PLAYING")
    }
}
